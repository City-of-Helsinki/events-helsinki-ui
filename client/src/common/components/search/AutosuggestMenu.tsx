import classNames from "classnames";
import React, { FunctionComponent } from "react";

import { ReactComponent as CloseIcon } from "../../../assets/icons/svg/close.svg";
import { formatMessage } from "../../translation/utils";
import { AutosuggestMenuItem } from "../../types";
import styles from "./autosuggestMenu.module.scss";

interface Props {
  items: AutosuggestMenuItem[];
  isOpen: boolean;
  onClose: () => void;
  onItemClick: (item: AutosuggestMenuItem) => void;
}

const AutosuggestMenu: FunctionComponent<Props> = ({
  items,
  isOpen,
  onClose,
  onItemClick
}) => {
  if (!isOpen) return null;
  return (
    <div className={styles.autosuggestMenu}>
      <div className={styles.title}>
        {formatMessage("commons.autosuggest.menu.title")}
        <button
          aria-label={formatMessage("commons.autosuggest.menu.ariaButtonClose")}
          className={styles.closeButton}
          onClick={onClose}
        >
          <CloseIcon />
        </button>
      </div>
      <ul className={styles.autosuggesItems}>
        {items.map((item, index) => {
          const handleClick = () => {
            onItemClick(item);
          };
          return (
            <li
              key={index}
              className={classNames(
                styles.autosuggesItem,
                styles[`autosuggestItem--${item.type}`]
              )}
            >
              <div className={styles.colorIndicator} />
              <div className={styles.icon} />
              <div className={styles.textWrapper}>
                <button onClick={handleClick}>
                  <div className={styles.text}>{item.text}</div>
                </button>
              </div>
            </li>
          );
        })}
      </ul>

      <div className={styles.info}>
        {formatMessage("commons.autosuggest.menu.info")}
      </div>
    </div>
  );
};

export default AutosuggestMenu;
