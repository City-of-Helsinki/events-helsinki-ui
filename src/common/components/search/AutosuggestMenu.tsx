import classNames from 'classnames';
import { IconCross } from 'hds-react';
import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import { AutosuggestMenuOption } from '../../types';
import styles from './autosuggestMenu.module.scss';

interface Props {
  focusedOption: number;
  isOpen: boolean;
  onClose: () => void;
  onOptionClick: (item: AutosuggestMenuOption) => void;
  options: AutosuggestMenuOption[];
}

const AutosuggestMenu: FunctionComponent<Props> = ({
  focusedOption,
  isOpen,
  onClose,
  onOptionClick,
  options,
}) => {
  const { t } = useTranslation();

  if (!isOpen) return null;

  return (
    <div className={styles.autosuggestMenu}>
      <div className={styles.title}>
        {t('commons.autosuggest.menu.title')}
        <button
          aria-label={t('commons.autosuggest.menu.ariaButtonClose')}
          className={styles.closeButton}
          onClick={onClose}
        >
          <IconCross />
        </button>
      </div>
      <ul className={styles.autosuggestOptions} role="listbox">
        {options.map((option, index) => {
          const handleClick = () => {
            onOptionClick(option);
          };

          return (
            <li
              key={index}
              className={classNames(
                styles.autosuggestOption,
                styles[`autosuggestOption--${option.type}`],
                {
                  [styles['autosuggestOption--isFocused']]:
                    focusedOption === index,
                }
              )}
              role="option"
              tabIndex={-1}
              onClick={handleClick}
              aria-selected={focusedOption === index}
            >
              <div className={styles.icon}></div>
              <div className={styles.textWrapper}>
                <div className={styles.text}>{option.text}</div>
              </div>
            </li>
          );
        })}
      </ul>

      <div className={styles.info}>{t('commons.autosuggest.menu.info')}</div>
    </div>
  );
};

export default AutosuggestMenu;
