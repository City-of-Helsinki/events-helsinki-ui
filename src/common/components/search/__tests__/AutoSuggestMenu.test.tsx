import * as React from "react";
import renderer from "react-test-renderer";

import AutosuggestMenu from "../AutosuggestMenu";

test("AutosuggestMenu matches snapshot", () => {
  const component = renderer.create(
    <AutosuggestMenu
      focusedOption={0}
      options={[{ text: "foo", type: "bar", value: "foo" }]}
      isOpen={true}
      onClose={() => {}}
      onOptionClick={() => {}}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
