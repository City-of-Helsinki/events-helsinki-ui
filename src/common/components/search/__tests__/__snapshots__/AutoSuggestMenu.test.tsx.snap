// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutosuggestMenu matches snapshot 1`] = `
<div
  class="autosuggestMenu"
>
  <div
    class="title"
  >
    Hakuehdotuksia
    <button
      aria-label="Sulje valikko"
      class="closeButton"
    >
      <svg
        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M0 0h24v24H0z"
          />
          <path
            d="M17.657 7.757L13.414 12l4.243 4.243-1.414 1.414L12 13.414l-4.243 4.243-1.414-1.414L10.586 12 6.343 7.757l1.414-1.414L12 10.586l4.243-4.243z"
            fill="currentColor"
          />
        </g>
      </svg>
    </button>
  </div>
  <ul
    class="autosuggestOptions"
    role="listbox"
  >
    <li
      aria-selected="true"
      class="autosuggestOption autosuggestOption--text autosuggestOption--isFocused"
      role="option"
    >
      <div
        class="icon"
      />
      <div
        class="textWrapper"
      >
        <div
          class="text"
        >
          foo
        </div>
      </div>
    </li>
  </ul>
  <div
    class="info"
  >
    Voit myös etsiä kirjoittamallasi hakusanalla
  </div>
</div>
`;
